src/snowflake/snowpark/mock/_plan.py

```diff
@@ -539,6 +563,282 @@ def handle_udf_expression(
         return res
 
 
+def handle_udaf_expression(
+    exp: FunctionExpression,
+    input_data: Union[TableEmulator, ColumnEmulator],
+    analyzer: "MockAnalyzer",
+    expr_to_alias: Dict[str, str],
+    current_row=None,
+):
+    udaf_registry = analyzer.session.udaf
+    udaf_name = exp.udf_name
+    udaf = udaf_registry.get_udaf(udaf_name)
+
+    with ImportContext(udaf_registry.get_udaf_imports(udaf_name)):
+        # Resolve handler callable
+        if type(udaf.handler) is tuple:
+            module_name, handler_name = udaf.func
+            exec(f"from {module_name} import {handler_name}")
+            udaf_handler = eval(handler_name)
+        else:
+            udaf_handler = udaf.handler
+
+        # Compute input data and validate typing
+        if len(exp.children) != len(udaf._input_types):
+            raise SnowparkLocalTestingException(
+                f"Expected {len(udaf._input_types)} arguments, but received {len(exp.children)}"
+            )
+
+        function_input = TableEmulator(index=input_data.index)
+        for child, expected_type in zip(exp.children, udaf._input_types):
+            col_name = analyzer.analyze(child, expr_to_alias)
+            column_data = calculate_expression(
+                child, input_data, analyzer, expr_to_alias
+            )
+
+            # Variant Data is often cast to specific python types when passed to a udf.
+            if isinstance(expected_type, VariantType):
+                column_data = column_data.apply(coerce_variant_input)
+
+            coerce_result = get_coerce_result_type(
+                column_data.sf_type, ColumnType(expected_type, False)
+            )
+            if coerce_result is None:
+                raise SnowparkLocalTestingException(
+                    f"UDAF received input type {column_data.sf_type.datatype} for column {child.name}, but expected input type of {expected_type}"
+                )
+
+            function_input[col_name] = cast_column_to(
+                column_data, ColumnType(expected_type, False)
+            )
+
+        try:
+            # Initiate some aggregation state (this is only needed for distributed compute).
+            some_agg_state = udaf_handler().aggregate_state
+            Agg = udaf_handler()
```

Comment:
Is Agg an instance or type? The type hinting doesn't give much clarity here so maybe a more verbose variable name or an explanatory comment would help.

Reply:
It's an instance of the UDAF class. https://docs.snowflake.com/en/developer-guide/snowpark/python/creating-udafs I clarified the code and renamed the variable to `AggregationAccumulator`